tests/ui/error-codes/E0606.stderr

```diff
@@ -2,10 +2,12 @@ error[E0606]: casting `&u8` as `u8` is invalid
   --> $DIR/E0606.rs:2:5
    |
 LL |     &0u8 as u8;
-   |     ----^^^^^^
-   |     |
-   |     cannot cast `&u8` as `u8`
-   |     help: dereference the expression: `*&0u8`
+   |     ^^^^^^^^^^
+   |
+help: dereference the expression
+   |
+LL |     *&0u8 as u8;
+   |     +
```

Comment:
You have access to the expression through `self.expr`. Would you mind special casing `hir::ExprKind::AddrOf` to suggest removing the `&` instead?